src/routes/process-user-vocabulary/components/AudioRecordOverlay.tsx

```diff
@@ -0,0 +1,201 @@
+import styled, { useTheme } from 'styled-components/native'
+
+import { CloseIcon, MicrophoneIcon } from '../../../../assets/images'
+import NotAuthorisedView from '../../../components/NotAuthorisedView'
+import { HeadingText } from '../../../components/text/Heading'
+import useGrantPermissions from '../../../hooks/useGrantPermissions'
+import { getLabels } from '../../../services/helpers'
+import { reportError } from '../../../services/sentry'
+
+const Container = styled.SafeAreaView`
+  flex: 1;
+  background-color: ${props => props.theme.colors.backgroundAccent};
+`
+
+const Icon = styled.Pressable`
+  align-self: flex-end;
+  margin: ${props => `${props.theme.spacings.xs} ${props.theme.spacings.md}`};
+`
+
+const RecordIcon = styled(Pressable)<{ isPressed: boolean }>`
+  width: ${hp('12%')}px;
+  height: ${hp('12%')}px;
+  align-self: center;
+  align-items: center;
+  justify-content: center;
+  border-radius: 100px;
+  background-color: ${props =>
+    props.isPressed ? props.theme.colors.audioRecordingActive : props.theme.colors.disabled};
+`
+
+const Content = styled.View`
+  margin-top: ${hp('12%')}px;
+  margin-bottom: ${hp('6%')}px;
+  flex: 1;
+  justify-content: space-between;
+`
+
+const Heading = styled(HeadingText)`
+  text-align: center;
+  font-size: ${props => props.theme.spacings.lg};
+`
+
+const HeadingContainer = styled.View`
+  min-height: ${hp('12%')}px;
+`
+
+const RecordingInfo = styled.Text`
+  font-size: ${props => props.theme.spacings.lg};
+  text-align: center;
+`
+
+const MeteringInfo = styled.View`
+  flex-direction: row;
+  align-self: center;
+  min-height: ${hp('12%')}px;
+`
+
+const MeteringBar = styled.View<{ height: number }>`
+  min-height: ${props => props.theme.spacings.xxs}
+  height: ${props => `${props.height * 2}`}px;
+  width: ${props => props.theme.spacings.xxs}
+  background-color: ${props => props.theme.colors.audioRecordingActive};
+  border-radius-: 50px;
+  align-self: center;
+  margin: 0 1px;
+`
+
+const InfoContainer = styled.View`
+  justify-content: flex-end;
+`
+
+interface AudioRecordOverlayProps {
+  onClose: () => void
+  setShowAudioRecordOverlay: Dispatch<SetStateAction<boolean>>
+  audioRecorderPlayer: AudioRecorderPlayer
+  setRecordingPath: Dispatch<SetStateAction<string | null>>
+  recordingPath: string | null
+  setRecordSeconds: Dispatch<SetStateAction<number>>
+}
+
+const recordingTimeInit = '00:00'
+const factor = 1000
+// The recording library does align the android power recording level to ios for metering as it's done in different libraries https://github.com/ziscloud/sound_recorder/blob/46544fc23b71e6f929b372fad0313c70b0301371/android/src/main/java/com/neuronbit/sound_recorder/SoundRecorderPlugin.java#L375
+const androidFactor = 0.25
+
+// Zero alignment of values with the minimum metering value
+const cleanUpMeteringResults = (meteringResults: number[]): number[] => {
+  const filteredResults = meteringResults.filter(el => el !== Math.min(...meteringResults))
```

Comment:
Why not do this? 
```suggestion
  const minimumMeteringValue = Math.min(...meteringResults)
  return meteringResults.map(el => el + Math.abs(minimumMeteringValue))
```

and why do you need to filter the minimum value? Shouldn't it be enough to normalize (the second step)?

Reply:
I'm gonna check your solution. the filtering is an android problem. When the records start there are dozens of minimum metering values without any sense and they make the metering bar very ugly. you can just log the metering results on android and gonna see why.

Reply:
checked and would keep my solution with the filtering. looks way better